### PR TITLE
feat: add agoric

### PR DESCRIPTION
**Pull request by chalabi2**
## 1. Summary
Features:
- Added Agoric
- Added search input to network selector

## 2.Type of change
- [X] New feature (non-breaking change which adds functionality)

## Summary by CodeRabbit

- **New Features**
	- Added support for the AGORIC network, including new endpoints and network configurations.
	- Introduced the `agoric` token in the assets page and updated token lists to include it.
	- Enhanced staking modal to handle new memo creation logic with base64 encoding.
  
- **Updates**
	- Updated chain ID settings for DYDX network to incorporate AGORIC chain ID.
	- Updated fee estimation hooks and queries for better accuracy and performance.

- **Bug Fixes**
	- Fixed denomination and zone listings to include `ubld` and AGORIC zone, ensuring compatibility with new network standards.

### CODE REVIEW

**Comment by chalabi2** [comment]
This might need a little debugging when the zone is actually added but should be good to go!

**Comment by joe-bowman** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>b0f9ff3</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://pr-1476.app-dev.quicksilver.zone'>https://pr-1476.app-dev.quicksilver.zone</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/quicksilver/quicksilver/2ngFMnkNuigBEEeg1Pinkovze4ui'>https://vercel.com/quicksilver/quicksilver/2ngFMnkNuigBEEeg1Pinkovze4ui</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/quicksilver-zone/quicksilver/actions/runs/8796323513)

**Comment by joe-bowman** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>7484365</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://pr-1476.app-dev.quicksilver.zone'>https://pr-1476.app-dev.quicksilver.zone</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/quicksilver/quicksilver/FvkNzw6o8Gd5PhPo57dAnax8o5Wt'>https://vercel.com/quicksilver/quicksilver/FvkNzw6o8Gd5PhPo57dAnax8o5Wt</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/quicksilver-zone/quicksilver/actions/runs/8796845224)

File: web-ui/.env [hunk 1]
```diff
@@ -32,4 +34,5 @@ NEXT_PUBLIC_SOMMELIER_CHAIN_ID=sommelier-3
 NEXT_PUBLIC_JUNO_CHAIN_ID=juno-1
 NEXT_PUBLIC_DYDX_CHAIN_ID=dydx-mainnet-1
 NEXT_PUBLIC_DYDX_CHAIN_ID=ssc-1
+NEXT_PUBLIC_DYDX_CHAIN_ID=agoric-3
```
Comment (joe-bowman) [inline web-ui/.env:37]:
```suggestion
NEXT_PUBLIC_DYDX_CHAIN_ID=dydx-mainnet-1
NEXT_PUBLIC_SAGA_CHAIN_ID=ssc-1
NEXT_PUBLIC_AGORIC_CHAIN_ID=agoric-3
```

File: web-ui/hooks/useFeeEstimation.ts [hunk 2]
```diff
@@ -21,7 +21,7 @@ export const useFeeEstimation = (chainName: string) => {
     if (!stargateClient) {
       throw new Error('getSigningStargateClient error');
     }
-
+    console.log(address)
```
Comment (joe-bowman) [inline web-ui/hooks/useFeeEstimation.ts:24]:
```suggestion
```

**Comment by joe-bowman** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>0578ebe</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://pr-1476.app-dev.quicksilver.zone'>https://pr-1476.app-dev.quicksilver.zone</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/quicksilver/quicksilver/FmQtatFxV2RvGuvJDZG2r2waZG2J'>https://vercel.com/quicksilver/quicksilver/FmQtatFxV2RvGuvJDZG2r2waZG2J</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/quicksilver-zone/quicksilver/actions/runs/8798737125)

**Comment by joe-bowman** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>d345737</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://pr-1476.app-dev.quicksilver.zone'>https://pr-1476.app-dev.quicksilver.zone</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/quicksilver/quicksilver/AkkdJBJH8zGzJRx2F4wgeeQK4Pqw'>https://vercel.com/quicksilver/quicksilver/AkkdJBJH8zGzJRx2F4wgeeQK4Pqw</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/quicksilver-zone/quicksilver/actions/runs/8798746868)

File: web-ui/state/chains/prod.ts [hunk 3]
```diff
@@ -20,7 +21,8 @@ export const ibcDenomWithdrawMapping = {
       qSOMM: 'ibc/EAF76AD1EEF7B16D167D87711FB26ABE881AC7D9F7E6D0CF313D5FA530417208',
       qJUNO: 'ibc/B4E18E61E1505C2F371B621E49B09E983F6A138F251A7B5286A6BDF739FD0D54',
       qDYDX: 'ibc/273C593E51ACE56F1F2BDB3E03A5CB81BB208B894BCAA642676A32C3454E8C27',
-      qSAGA: 'ibc/F2D400F2728E9DA06EAE2AFAB289931A69EDDA5A661578C66A3177EDFE3C0D13'
+      qSAGA: 'ibc/F2D400F2728E9DA06EAE2AFAB289931A69EDDA5A661578C66A3177EDFE3C0D13',
+      qBLD: ''
```
Comment (joe-bowman) [inline web-ui/state/chains/prod.ts:25]:
```suggestion
      qBLD: 'ibc/C1C106D915C8E8C59E5DC69BF30FEF64729A6F788060B184C86A315DBB762EF7'
```

File: web-ui/state/chains/prod.ts [hunk 4]
```diff
@@ -30,7 +32,8 @@ export const ibcDenomWithdrawMapping = {
       qSOMM: 'ibc/ACF9DA139FE5BC8F95AC4A12B0B6D7710274DEDAC57284B881BEE1896F40642D',
       qJUNO: 'ibc/CA0BEF2524A37205009210EFCFB09585FBA9648C5F065FA078944A5C6704E8DC',
       qDYDX: 'ibc/41F3C94FAB3FB2D6D2B1F130A78697B07D729D1F50DA132C18F7963413A2DCF6',
-      qSAGA: 'ibc/9B4BDA7382D0CF8C48A9D7496449D626DDF99AF640325978B5BD1AD4A9ED274C'
+      qSAGA: 'ibc/9B4BDA7382D0CF8C48A9D7496449D626DDF99AF640325978B5BD1AD4A9ED274C',
+      qBLD: ''
```
Comment (joe-bowman) [inline web-ui/state/chains/prod.ts:36]:
```suggestion
      qBLD: 'ibc/3DD2D3E1BA055899A6E5E76BD2675F462988880EBE92216CD36071335ADE699C'
```

**Review by joe-bowman — DISMISSED** [review]
lgtm

**Comment by joe-bowman** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>2574063</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://pr-1476.app-dev.quicksilver.zone'>https://pr-1476.app-dev.quicksilver.zone</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/quicksilver/quicksilver/DQVWJUeAsarNBQ2kUUrfVbz9xn84'>https://vercel.com/quicksilver/quicksilver/DQVWJUeAsarNBQ2kUUrfVbz9xn84</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/quicksilver-zone/quicksilver/actions/runs/8799320401)

**Comment by joe-bowman** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>98ee317</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://pr-1476.app-dev.quicksilver.zone'>https://pr-1476.app-dev.quicksilver.zone</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/quicksilver/quicksilver/6UkJhPGVKGo85JAXjwQ8sQGuvvjL'>https://vercel.com/quicksilver/quicksilver/6UkJhPGVKGo85JAXjwQ8sQGuvvjL</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/quicksilver-zone/quicksilver/actions/runs/8801564285)

**Comment by joe-bowman** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>ef371b2</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://pr-1476.app-dev.quicksilver.zone'>https://pr-1476.app-dev.quicksilver.zone</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/quicksilver/quicksilver/7F3X7RLYpwrjpxTb1dPF4zTHt7eM'>https://vercel.com/quicksilver/quicksilver/7F3X7RLYpwrjpxTb1dPF4zTHt7eM</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/quicksilver-zone/quicksilver/actions/runs/8804941722)

**Comment by joe-bowman** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>523d1ad</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://pr-1476.app-dev.quicksilver.zone'>https://pr-1476.app-dev.quicksilver.zone</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/quicksilver/quicksilver/ApE26w9j3YxPeRPsP9h3HoCM66DX'>https://vercel.com/quicksilver/quicksilver/ApE26w9j3YxPeRPsP9h3HoCM66DX</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/quicksilver-zone/quicksilver/actions/runs/8811433092)

File: web-ui/components/Staking/modals/stakingProcessModal.tsx [hunk 5]
```diff
@@ -209,33 +203,58 @@ export const StakingProcessModal: React.FC<StakingModalProps> = ({ isOpen, onClo
...
+    return Buffer.concat([Buffer.from([0x00, 0x14]), wordsUint8Array]);
+  };
+
+  // Create the memo
   let memoBuffer = Buffer.alloc(0);
 
+  // Example address and intents
+
+  // Add mapped account to memo
+  memoBuffer = Buffer.concat([memoBuffer, addMappedAccount(address ?? 'quick1uwqjtgjhjctjc45ugy7ev5prprhehc7wje9uwh')]);
```
Comment (joe-bowman) [inline web-ui/components/Staking/modals/stakingProcessModal.tsx:243]: Let's remove the default so you don't get everyone's qAssets :D

File: web-ui/components/Staking/modals/stakingProcessModal.tsx [hunk 6]
```diff
@@ -209,33 +203,58 @@ export const StakingProcessModal: React.FC<StakingModalProps> = ({ isOpen, onClo
...
+
+  // Add validator details to memo
   if (intents.length > 0) {
     intents.forEach((val) => {
       memoBuffer = Buffer.concat([memoBuffer, addValidator(val.address, val.intent)]);
     });
+
+    // Add header with the memo buffer length
     memoBuffer = Buffer.concat([Buffer.from([0x02, memoBuffer.length]), memoBuffer]);
   }
```
Comment (joe-bowman) [inline web-ui/components/Staking/modals/stakingProcessModal.tsx:254]: prepend / append the addMappedAccounts to the memoBuffer here.

Comment (joe-bowman) [reply web-ui/components/Staking/modals/stakingProcessModal.tsx:254]: Conditionally on zone.is_118 being false.

File: web-ui/components/Staking/modals/stakingProcessModal.tsx (hunk 5 above)
Comment (chalabi2) [reply web-ui/components/Staking/modals/stakingProcessModal.tsx:243]: lol not a good default huh... This was a hacky solution to ensure `addMappedAccounts` props are never undefined.

**Comment by joe-bowman** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>17bc92d</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://pr-1476.app-dev.quicksilver.zone'>https://pr-1476.app-dev.quicksilver.zone</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/quicksilver/quicksilver/5nE3uWfbzCuLSsXTvN1TtRdVEHij'>https://vercel.com/quicksilver/quicksilver/5nE3uWfbzCuLSsXTvN1TtRdVEHij</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/quicksilver-zone/quicksilver/actions/runs/8822380044)

File: web-ui/components/Staking/modals/stakingProcessModal.tsx (hunk 5 above)
Comment (joe-bowman) [reply web-ui/components/Staking/modals/stakingProcessModal.tsx]: So we only want to populate it for non-118 zones; and if a mapped account in not submitted and _is_ needed (non-118, and no previously used mapped account, then we send the tokens back to the origin account.

File: web-ui/components/Staking/modals/stakingProcessModal.tsx [hunk 7]
```diff
@@ -209,32 +203,54 @@ export const StakingProcessModal: React.FC<StakingModalProps> = ({ isOpen, onClo
...
+  // Add validator details to memo
   if (intents.length > 0) {
     intents.forEach((val) => {
       memoBuffer = Buffer.concat([memoBuffer, addValidator(val.address, val.intent)]);
     });
+    // Add mapped account to memo
+    if (zone?.chainId === 'agoric-3' && address) {
+      memoBuffer = Buffer.concat([memoBuffer, addMappedAccount(address)]);
+    }
+    // Add header with the memo buffer length
```
Comment (joe-bowman) [inline web-ui/components/Staking/modals/stakingProcessModal.tsx:248]: This needs to be after line 249. The 0x02 <length> prefix pertains to the intents field.

Lines 244-248 should not be inside the `if (intents.length > 0) {` condition.

**Comment by joe-bowman** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>6b3ad3f</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://pr-1476.app-dev.quicksilver.zone'>https://pr-1476.app-dev.quicksilver.zone</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/quicksilver/quicksilver/9XWw1YkUktpU2e2j1jF2P5dQoF5z'>https://vercel.com/quicksilver/quicksilver/9XWw1YkUktpU2e2j1jF2P5dQoF5z</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/quicksilver-zone/quicksilver/actions/runs/8833265316)

File: web-ui/components/Staking/modals/stakingProcessModal.tsx (hunk 7 above)
Comment (chalabi2) [reply web-ui/components/Staking/modals/stakingProcessModal.tsx]: apologies for the oversight!

**Pull request closed without merging**